include/zephyr/drivers/counter.h

```diff
@@ -35,6 +35,18 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Define the typedef to be 32b or 64b for ticks
+ *
+ * It is still okay for a driver or application to still use `uint32_t` or
+ * `uint64_t` if it is to always expected to be that.
```

Comment:
It seems not. CI reports build failures on non matching handler prototypes/signatures:
```
/__w/zephyr/zephyr/drivers/counter/counter_esp32_tmr.c:320:22: error: initialization of 'int (*)(const struct device *, counter_ticks_t *)' {aka 'int (*)(const struct device *, long long unsigned int *)'} from incompatible pointer type 'int (*)(const struct device *, uint32_t *)' {aka 'int (*)(const struct device *, unsigned int *)'} [-Werror=incompatible-pointer-types]
  320 |         .get_value = counter_esp32_get_value,
      |                      ^~~~~~~~~~~~~~~~~~~~~~~

Reply:
likely that was just an issue with the rebase I did